shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/app/ClientProducerImpl.java

```diff
@@ -0,0 +1,19 @@
+package com.redhat.service.bridge.shard.operator.app;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+@Singleton
+public class ClientProducerImpl implements ClientProducer {
```

Comment:
> Why this class? Isn't `OpenShiftClient` available as CDI bean? And what if the shard is running on k8s instead of OpenShift?

@danielezonca `OpenShiftClient extends KubernetesClient` so we make use of polymorphism to do the magic. The "general purpose" beans will use 

```
@Inject
KubernetesClient
```
So they use the client as a standard `KubernetesClient`. 

The platform dependent beans have a producer like 

```
    @Inject
    OpenShiftClient client;
    
    @Produces
    public NetworkingService getService() {
        if (Platform.OPENSHIFT.equals(platformConfigProvider.getPlatform())) {
            return new OpenshiftNetworkingService(client, templateProvider);
        }
        return new KubernetesNetworkingService(client, templateProvider);
    }
```

And
```
    public KubernetesNetworkingService(KubernetesClient client, TemplateProvider templateProvider) {
        this.client = client;
        this.templateProvider = templateProvider;
    }
```
and 

```
    public OpenshiftNetworkingService(OpenShiftClient client, TemplateProvider templateProvider) {
        this.client = client;
        this.templateProvider = templateProvider;
    }
```

Looking now if there is `OpenShiftClient` already available

Reply:
Yes there is already a CDI bean for `OpenShiftClient` and it seems it works fine. I've removed these classes